pkg/lib/cockpit.js

```diff
@@ -2613,11 +2616,17 @@
     Object.defineProperty(cockpit, "location", {
         enumerable: true,
         get: function() {
-            if (!last_loc || last_loc.href !== get_window_location_hash())
+            if (!last_loc || last_loc.href !== get_window_location_hash()) {
                 last_loc = new Location();
+                if (window.debugging?.includes("service"))
+                    console.log("XXX cockpit.location.get; window.location.hash", JSON.stringify(window.location.hash), "creating new last_loc");
+            } else
+                if (window.debugging?.includes("service"))
+                    console.log("XXX cockpit.location.get; window.location.hash", JSON.stringify(window.location.hash), "unchanged last_loc");
             return last_loc;
         },
         set: function(v) {
+            console.log("YYY cockpit.location.set", JSON.stringify(v));
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18422-20230307-065433-1c06e6c8-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)